src/modules/sync/validation.ts

```diff
@@ -0,0 +1,64 @@
+import chalk from "chalk";
+import * as fs from "fs/promises";
+import * as path from "path";
+
+import { contentTypesFileName, contentTypeSnippetsFileName, taxonomiesFileName } from "./constants/filename.js";
+import { FileContentModel } from "./types/fileContentModel.js";
+import { extractTerms } from "./utils/taxonomyGroupHelpers.js";
+
+export const validateContentFolder = async (folderPath: string) => {
+  const stats = await fs.stat(folderPath);
+
+  if (!stats.isDirectory()) {
+    return [`The provided path ${chalk.yellow(folderPath)} is not a valid content model folder`];
+  }
+
+  const promiseArray = await Promise.all(
+    [contentTypesFileName, contentTypeSnippetsFileName, taxonomiesFileName].map(filename =>
+      fs.stat(path.resolve(folderPath, filename)).catch(e => {
+        return `Could not find required file ${chalk.yellow(filename)} due to ${chalk.red(e)}`;
+      })
+    ),
+  );
+
+  return promiseArray.filter((p): p is string => typeof p === "string");
```

Comment:
It's not really a promise array, is it? You merged the promises into one and awaited the resulting promise.